### ISSUE TITLE
compiling gives error on Linux

### ISSUE THREAD

**Issue by tissatussa** [A1exL1ang/Superultra#1]
i just discovered your engine and i tried to compile it on Linux but i didn't succeed .. i think i solved SOME errors, by correcting your sources, but other errors are unclear to me .. i'm not into C++ too much, i found some answers on StackOverflow and it finally resulted in a binary of size about 5 Mb, which seems to run fine, but ..: it accepts the `uci` command, even `ucinewgame` and `position startpos`, but then after `go infinite` (a valid UCI command) my terminal output shows _Segmentation fault (core dumped)_ ..

here are some code changes i tried to solve it.

regarding the makefile : `CXXFLAGS := -Wl,--stack,8589934592` into `CXXFLAGS := -Wl,-z,stack-size=8589934592` (?!)
changing `-std=gnu++2a` into `-std=c++2a` (?!)
compiling by `make build=debug` (?!)

i corrected some `std::min` and `std::max` in `search.cpp` and `nnue.cpp`, eg. like this :
`Depth R = 3 + (depth / 3) + std::min((ss->staticEval - beta) / 200, 3);`
into
`Depth R = 3 + (depth / 3) + std::min(static_cast<int>((ss->staticEval - beta) / 200), 3);`

according to some forum answers the two min / max parameters should have same type, thus i tried this casting ..

i also fixed this simple error :

```
nnue.cpp:5:1: note: ‘memcpy’ is defined in header ‘<cstring>’; did you forget to ‘#include <cstring>’?
    4 | #include "nnue.h"
  +++ |+#include <cstring>
```

..by adding `#include <cstring>` on top of `nnue.cpp`.

i hope this info and suggestions can help you correct your code to make it work on Linux ?
my coding might be primitive and wrong, but i guess you get an idea how to fix it properly.

**Comment by A1exL1ang** [comment]
Thank you so much! My next release should address these issues and other bugs.

**Comment by tissatussa** [comment]
hi, i just discovered you changed some code in the main branch but i didn't receive an email notification of it, although i've set these ON .. nevermind

i tried to compile again but i didn't succeed yet .. i can fix all mentioned errors during `make`, mainly by adding `static_cast<int>` .. you didn't change these ?! On Windows it's OK, i guess, but when it's also OK with a cast, why not add those ?

the main issue is the stack argument : i can not find any setting & option for Linux and compiling without the stack option gives an invalid binary (core dumped or so..)

**Comment by tissatussa** [comment]
@A1exL1ang no, i can not compile your v2.0 code, some of the mentioned errors remain and i still get _Segmentation fault (core dumped)_ ..

btw. i discovered the number 8589934592 is 8 Gb .. that's the full memory size of my notebook, so i adjusted this number into 536870912 (0.5 Gb) but it didn't help .. (i used "-Wl,-z,stack-size=536870912").

**Comment by tissatussa** [comment]
v2.1 compiles very well on Linux (no errors and warnings) and plays a fine game in CuteChess GUI !
